src/parquet/arrow/reader.cc

```diff
@@ -903,7 +924,22 @@ Status ColumnReader::Impl::NextBatch(int batch_size, std::shared_ptr<Array>* out
           return TypedReadBatch<::arrow::TimestampType, Int96Type>(batch_size, out);
           break;
         default:
-          return Status::NotImplemented("TimeUnit not supported");
+          return UnsupportedTimeUnit(timestamp_type->unit);
+      }
+      break;
+    }
+    case ::arrow::Type::TIME: {
+      ::arrow::TimeType* time_type =
+          static_cast<::arrow::TimeType*>(field_->type.get());
+      switch (time_type->unit) {
+        case ::arrow::TimeUnit::MILLI:
+          return TypedReadBatch<::arrow::TimeType, Int32Type>(batch_size, out);
+          break;
+        case ::arrow::TimeUnit::NANO:
+          return TypedReadBatch<::arrow::TimeType, Int64Type>(batch_size, out);
+          break;
+        default:
+          return UnsupportedTimeUnit(time_type->unit);
```

Comment:
The time type is undergoing discussion in https://issues.apache.org/jira/browse/ARROW-617 so most likely these will have to be revisited after the time type is finalized in the Arrow format.